src/main/java/io/zold/api/RtTransaction.java

```diff
@@ -62,13 +67,29 @@
         this.transaction = trnsct;
     }
 
-    // @todo #15:30min Implement id() by parsing the string representation
-    //  of transaction according to the pattern, described in the white
-    //  paper. Replace relevant test case with actual tests.
     @Override
     @SuppressWarnings("PMD.ShortMethodName")
-    public long id() {
-        throw new UnsupportedOperationException("id() not yet implemented");
+    public long id() throws IOException {
+        try {
+            final String ident = new ItemAt<>(
+                new SplitText(this.transaction, ";").iterator(),
+                //@checkstyle MagicNumber (12 lines)
+                0
+            ).value().asString();
+            if (!RtTransaction.IDENT.matcher(ident).matches()) {
+                throw new IllegalArgumentException(
+                    String.format(
+                        // @checkstyle LineLength (1 line)
+                        "Invalid ID '%s' expecting 64-bit signed hex string with 4 symbols",
+                        ident
+                    )
+                );
+            }
+            return Integer.parseUnsignedInt(ident, 16);
```

Comment:
@carlosmiranda Shouldn't it be `Long.parseUnsignedLong`?

Reply:
@victornoel the ID field is actually smaller than `int` so `Integer.parseUnsignedInt` is enough.

Reply:
@carlosmiranda in that case I think we should change the contract of the method `id()` in the interface to `int`.